common/src/main/java/io/netty/util/internal/ObjectCleaner.java

```diff
@@ -41,17 +47,21 @@ public void run() {
                 // Keep on processing as long as the LIVE_SET is not empty and once it becomes empty
                 // See if we can let this thread complete.
                 while (!LIVE_SET.isEmpty()) {
+                    final AutomaticCleanerReference reference;
                     try {
-                        AutomaticCleanerReference reference =
-                                (AutomaticCleanerReference) REFERENCE_QUEUE.remove();
-                        try {
-                            reference.cleanup();
-                        } finally {
-                            LIVE_SET.remove(reference);
-                        }
+                        reference = (AutomaticCleanerReference) REFERENCE_QUEUE.remove(REFERENCE_QUEUE_POLL_TIMEOUT_MS);
                     } catch (InterruptedException ex) {
                         // Just consume and move on
                         interrupted = true;
+                        continue;
+                    }
+                    if (reference != null) {
+                        try {
+                            reference.cleanup();
+                        } catch (Throwable ignored) {
+                            // ignore exceptions, and don't log in case the logger throws an exception.
```

Comment:
I think we should try to log and just guard the logger itself. So there will be at least a chance the user see the exception

Reply:
We can log but here are potential motivation to not log:
- the JDK also does the same:
https://github.com/netroby/jdk9-dev/blob/master/jdk/src/java.base/share/classes/jdk/internal/ref/CleanerImpl.java#L153
- the logger may also block or have other side effects.

WDYT?